gcpim.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	imtypes "cloud-android-orchestration/api/instancemanager/v1"
+	compute "cloud.google.com/go/compute/apiv1"
+	"context"
+	"errors"
+	"github.com/google/uuid"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+const (
+	// TODO(b/220891296): Make this configurable
+	projectId   = "google.com:cloud-android-jemoreira"
+	sourceImage = "projects/cloud-android-releases/global/images/cuttlefish-google-vsoc-0-9-21"
+	networkName = "projects/cloud-android-jemoreira/global/networks/default"
+)
+
+const (
+	namePrefix  = "cf-"
+	labelPrefix = "cf-"
+)
+
+var newUUIDString = func() string {
+	return uuid.New().String()
+}
+
+// GCP implementation of the instance manager.
+type GCPIM struct {
+	client *compute.InstancesClient
+}
+
+func NewGCPIM(client *compute.InstancesClient) *GCPIM {
+	result := &GCPIM{
+		client: client,
+	}
+	return result
+}
+
+func (m *GCPIM) DeviceFromId(name string, _ UserInfo) (DeviceDesc, error) {
+	return DeviceDesc{"127.0.0.1", "cvd-1"}, nil
+}
+
+func (m *GCPIM) InsertHost(zone string, req *imtypes.InsertHostRequest, user UserInfo) (*imtypes.Operation, error) {
+	if err := validateRequest(req); err != nil {
+		return nil, err
+	}
+	ctx := context.Background()
+	computeReq := &computepb.InsertInstanceRequest{
+		Project: projectId,
+		Zone:    zone,
+		InstanceResource: &computepb.Instance{
+			Name:           proto.String(namePrefix + newUUIDString()),
+			MachineType:    proto.String(req.HostInfo.GCP.MachineType),
+			MinCpuPlatform: proto.String(req.HostInfo.GCP.MinCPUPlatform),
+			Disks: []*computepb.AttachedDisk{
+				{
+					InitializeParams: &computepb.AttachedDiskInitializeParams{
+						DiskSizeGb:  proto.Int64(int64(req.HostInfo.GCP.DiskSizeGB)),
+						SourceImage: proto.String("projects/cloud-android-releases/global/images/cuttlefish-google-vsoc-0-9-21"),
+					},
+					Boot: proto.Bool(true),
+				},
+			},
+			NetworkInterfaces: []*computepb.NetworkInterface{
+				{
+					Name: proto.String(networkName),
+					AccessConfigs: []*computepb.AccessConfig{
+						{
+							Name: proto.String("External NAT"),
+							Type: proto.String(computepb.AccessConfig_ONE_TO_ONE_NAT.String()),
+						},
+					},
+				},
+			},
+			Labels: map[string]string{
+				labelPrefix + "creator":  user.Username(),
```

Comment:
add the label used by acloud too (created_by), it's ok to have both creator and created_by, though we'd probably be better sticking with acloud's name only.

Reply:
done, kept both looking forward to remove the the acloud one once deprecated and have all the labels with a specific prefix, `cvd` in this case.